shim/linux/src/entry.c

```diff
@@ -66,6 +86,28 @@ dispatch_system_kvm_check_extension(void)
 static long
 dispatch_system_kvm_create_vm(void)
 {
+    char vmname[22];
+    struct shim_vm_t *vm =
+        (struct shim_vm_t *)vmalloc(sizeof(struct shim_vm_t));
+    if (NULL == vm) {
+        goto vm_free;
+    }
+    if (handle_system_kvm_create_vm(&vm->vmid)) {
+        goto vm_free;
+    }
+    snprintf(vmname, sizeof(vmname), "kvm-vm:%d", vm->vmid);
+    int32_t fd = anon_inode_getfd(vmname, &fops_vm, vm, O_RDWR | O_CLOEXEC);
```

Comment:
I am surprised this compiled. In C you cannot create a variable in the middle of a function. Did you compile this code? If it is compiling, I would still move the variable declaration to the top.